File: gestionDesTournois Backend/src/main/java/org/example/football/entites/Tournoi.java
package org.example.football.entites;

import jakarta.persistence.*;
import lombok.*;



import java.time.LocalDate;
import java.util.List;
import org.example.football.Enum.StatutTournois;
import org.example.football.Enum.TypeTournoi;
@Entity
@Table(name = "tournoi")
@Data
@NoArgsConstructor
@AllArgsConstructor
@ToString(exclude = {"matchs", "participations"})
@EqualsAndHashCode(exclude = {"matchs", "participations"})
public class Tournoi {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String nom;

    private LocalDate dateDebut;
    private LocalDate dateFin;

    private int nbEquipes;

    private int nbParticipations;

    private int nbJoueurSurTerrain;


    @Enumerated(EnumType.STRING)
    private TypeTournoi typeTournoi;

    @Enumerated(EnumType.STRING)
    private StatutTournois    statut = StatutTournois.EN_ATTENTE;

    private boolean ouvert = true;

    private boolean estLance = false;





    @OneToMany(mappedBy = "tournoi", cascade = CascadeType.ALL)
    private List<Match> matchs;

    @OneToMany(mappedBy = "tournoi", cascade = CascadeType.ALL)
    private List<Participation> participations;
}

File: gestion-tournoi-frontend/src/app/interface/tournoi.ts
// tournoi-dto.ts
export interface TournoiDTO {
  id: number;
  nom: string;
  dateDebut: string;
  dateFin: string;
  nbEquipes: number;
  nbParticipations: number;
  typeTournoi: string;
  statut: string;
  ouvert: boolean;
  estLance: boolean;
  nbJoueurSurTerrain:number;
}


export interface TournoiCreateDTO {
  nom: string;
  dateDebut: string;
   nbJoueurSurTerrain:number;
  dateFin: string;
  nbEquipes: number;
  typeTournoi: string;
}